src/Mod/Sketcher/Gui/DrawSketchHandlerArc.h

```diff
@@ -887,6 +911,57 @@ void DSHArcController::addConstraints()
     }
 }
 
+template<>
+void DSHArcController::doConstructionMethodChanged()
+{
+    handler->updateHint();
+}
+
+// Static member definitions
+const Gui::InputHint DrawSketchHandlerArc::SWITCH_MODE_HINT = {QObject::tr("%1 switch mode"),
+                                                               {Gui::InputHint::UserInput::KeyM}};
+
+const DrawSketchHandlerArc::HintTable DrawSketchHandlerArc::ARC_HINT_TABLE = {
+    // Center method
+    {ConstructionMethod::Center,
+     SelectMode::SeekFirst,
+     {{QObject::tr("%1 pick arc center"), {Gui::InputHint::UserInput::MouseLeft}},
+      SWITCH_MODE_HINT}},
+    {ConstructionMethod::Center,
+     SelectMode::SeekSecond,
+     {{QObject::tr("%1 pick arc start point"), {Gui::InputHint::UserInput::MouseLeft}},
+      SWITCH_MODE_HINT}},
+    {ConstructionMethod::Center,
+     SelectMode::SeekThird,
+     {{QObject::tr("%1 pick arc end point"), {Gui::InputHint::UserInput::MouseLeft}},
+      SWITCH_MODE_HINT}},
+
+    // ThreeRim method
+    {ConstructionMethod::ThreeRim,
+     SelectMode::SeekFirst,
+     {{QObject::tr("%1 pick first arc point"), {Gui::InputHint::UserInput::MouseLeft}},
+      SWITCH_MODE_HINT}},
+    {ConstructionMethod::ThreeRim,
+     SelectMode::SeekSecond,
+     {{QObject::tr("%1 pick second arc point"), {Gui::InputHint::UserInput::MouseLeft}},
+      SWITCH_MODE_HINT}},
+    {ConstructionMethod::ThreeRim,
+     SelectMode::SeekThird,
+     {{QObject::tr("%1 pick third arc point"), {Gui::InputHint::UserInput::MouseLeft}},
+      SWITCH_MODE_HINT}}};
+
+std::list<Gui::InputHint> DrawSketchHandlerArc::lookupArcHints(ConstructionMethod method,
+                                                               SelectMode state)
+{
+    auto it = std::find_if(ARC_HINT_TABLE.begin(),
+                           ARC_HINT_TABLE.end(),
+                           [method, state](const HintEntry& entry) {
+                               return entry.method == method && entry.state == state;
+                           });
+
+    return (it != ARC_HINT_TABLE.end()) ? it->hints : std::list<Gui::InputHint> {};
+}

```

Comment:
That code is much harder to follow - I think that it would be better to just create few methods - one per construction method and based on the construction method call appropriate one.